_posts/2018-08-04-pfsense-as-hvm-on-pc-engines-apu2.md

```diff
@@ -0,0 +1,562 @@
+super-block backups (for fsck_ffs -b #) at:
+ 192, 4352, 8512, 12672
+
+Welcome to pfSense!
+
+Please choose the appropriate terminal type for your system.
+Common console types are:
+   ansi     Standard ANSI terminal
+   vt100    VT100 or compatible terminal
+   xterm    xterm terminal emulator (or compatible)
+   cons25w  cons25w terminal
+
+Console type [vt100]:
+```
+
+Unfortunately pfSense had problem getting DHCP offer and didn't configure IP
+address - I tried to figure out what is wrong but my BSD-fu is low. I will
+probably attack that problem later.
+
+# Xen debian.cfg
+
+```
+name = "debian-9.5.0"
+builder = "hvm"
+vcpus = 2
+memory = 2048
+pci = [ '02:00.0'  ]
+disk=[ '/root/debian-9.5.0-amd64-netinst.iso,,hdc,cdrom', '/dev/vg0/debian,,hdb,rw'  ]
+vnc=1
+vnclisten='apu2_ip_addr'
+boot='d'
+```
+
+Of course you have to replace `apu2_ip_add` with correct IP. After `xl create
+debian.cfg` you can run VNC (tightvnc worked for me) and proceed with
+installation.
+
+## PCI passthrough in Debian
+
+Below screenshot show device `02:00.0`, which is apu2 middle NIC,
+passed-through to VM.
+
+![Debian lspci](https://3mdeb.com/wp-content/uploads/2017/07/debian-9.5.0-hvm-pci-passthrough.png)
+
+PCI passthrough on Debian worked without any issue DHCP offer was received
+correctly and I could proceed with performance checks.
+
+# Speedtest
+
+Simplest possible test is comparison of through put between eth0 and eth1.
+First is connected directly to our company switch and second connects pfSense
+HVM using PCI passthrough.
+
+I used `speedtest-cli v2.0.2`.
+
+Results for apu2 Dom0:
+
+```
+(speedtest-venv) root@apu2:~# speedtest-cli 
+Retrieving speedtest.net configuration...
+Testing from Vectra Broadband (109.241.231.46)...
+Retrieving speedtest.net server list...
+Selecting best server based on ping...
+Hosted by Volta Communications Sp. z o.o (Gdansk) [2.28 km]: 36.105 ms
+Testing download speed................................................................................
+Download: 81.67 Mbit/s
+Testing upload speed................................................................................................
+Upload: 15.38 Mbit/s
+```
+
+Results for Debian HVM with NIC PCI passthrough:
+
+![Debian HVM speedtest-cli](https://3mdeb.com/wp-content/uploads/2017/07/speedtest-cli-debian-hvm.png)
+
+# iperf
+
+Below results are for very simple LAN connection `apu3 -> switch -> apu2`:
+
+```
+(speedtest-venv) root@apu2:~# iperf -s -B 192.168.3.101
+------------------------------------------------------------
+Server listening on TCP port 5001
+Binding to local address 192.168.3.101
+TCP window size: 85.3 KByte (default)
+------------------------------------------------------------
+[  4] local 192.168.3.101 port 5001 connected with 192.168.3.102 port 34004
+[ ID] Interval       Transfer     Bandwidth
+[  4]  0.0-10.0 sec  1.10 GBytes   941 Mbits/sec
+```
+
+Unfortunately our switch is probably not well suited for testing 1GbE. Those
```

Comment:
It turns out that switches are 1Gb. It will probably be a problem on pfSense side. If yes, this sentence should be removed.

Reply:
@miczyg1 this TBD, we should try to figure out what is going on here.